apps/openmw/mwmechanics/aicombat.cpp

```diff
@@ -490,6 +490,28 @@ namespace MWMechanics
 
     void AiCombatStorage::startCombatMove(bool isDistantCombat, float distToTarget, float rangeAttack, const MWWorld::Ptr& actor, const MWWorld::Ptr& target)
     {
+        // get the range of the target's weapon
+        float rangeAttackOfTarget = 0.f;
+        MWWorld::Ptr targetWeapon = MWWorld::Ptr();
+        const MWWorld::Class& targetClass = target.getClass();
+
+        if (targetClass.hasInventoryStore(target))
+        {
+            MWMechanics::WeaponType weapType = WeapType_None;
+            MWWorld::ContainerStoreIterator weaponSlot =
+                MWMechanics::getActiveWeapon(targetClass.getCreatureStats(target), targetClass.getInventoryStore(target), &weapType);
+            if (weapType != WeapType_PickProbe && weapType != WeapType_Spell && weapType != WeapType_None && weapType != WeapType_HandToHand)
+                targetWeapon = *weaponSlot;
+        }
+
+        std::shared_ptr<Action> targetWeaponAction(new ActionWeapon(targetWeapon));
```

Comment:
Should this be changed to a `unique_ptr` since it is not being shared with anything?

Reply:
Also, if it should be a ```shared_ptr``` for some reason, the proper way to instantiate it is with ```make_shared```:
```
std::shared_ptr<Action> targetWeaponAction = std::make_shared<ActionWeapon>(targetWeapon);
```

Reply:
Looking more closely at this, why does this need to be a pointer at all? At first I assumed that it needed to be so that it could call `getCombatRange()` polymorphically, but it looks like it can only be an `ActionWeapon`.